Add processes (#68)
meilisearch/integration-guides: guides/integration-tool-release.md

# Releasing an Integration Tool

This guide is about releasing integration tools and has nothing to do with MeiliSearch releases that are handled during the [pre-release week](./pre-release-week.md).

> ⚠️ Integration tool releases are public and are certainly among our most consulted contents with the READMEs.<br>
> Indeed, users reading the release changelogs are mostly users wanting to upgrade their dependencies, so they might consult the release description really carefully.
>
> **The Integration team should keep this in mind during the release process.**

### Why Doing a Release?

First of all, you need to check the release would include user-oriented changes. A release is not always needed after merging PRs.<br>
Don't do a release without any purpose, or if only the text files, tests, or CI have been updated.<br>

> 📢 **A release should be done only when there are changes concerning the code usage.**

Reviewer: ```suggestion
📢 **A release should be done only when there are changes concerning the code.**
```

Not only code usage itself. Exaple: optimization. Usage is the same, yet should be released?

Author: Real performance improvement -> yes. Refacto of the code that does not impact the usage -> no.
So this is only code usage for me. WDYT?

Author: The "code" is a too wide term. I can change the name of an internal variable. The code change, but the usage does not.

Author: ```suggestion
📢 **A release should be done only when there are changes concerning the package usage.**
```

Author: ```suggestion
📢 **A release should be done only when there are changes concerning the usage of the package.**
```

Reviewer: Agreed. It would make it just too vage! 👍🏽

Author: ```suggestion
📢 **A release should be done only when there are changes concerning the usage of the library.**
```

Author: WDYT of my new suggestions @eskombro? 😄

Reviewer: Ugh, I don't know. I don't like to call it a "package" or a "library", because I always think that this applies to other kind of tools. Yet using "tool" or "integration" is too general, unclear, and even boring :)

I kind of wish I hadn't started this discussion 😂

I'd go back to the "code usage" :)

Author: Got it! I let it as it is now. If Charlotte thinks this is weird, we re-discuss this! (She will read all of these guides when coming back from holidays).

### And Next?

Follow the steps in the `Release Process` section in the `CONTRIBUTING.md` file of the repository carefully. These steps should be precise; if it's not, please open an issue on the related repository.

During these steps, you might publish a release via the GitHub interface and so provide a description with the release changelogs.<br>
The [Release Drafter tool](./release-drafter.md), integrated into most repositories, helps you provide a kind of structure, but please, feel free to add final modifications to ensure these changelogs are:
- **accurate**: describe what changed, with links to the PRs, and the consequences of the changes if needed. Also, check all the breaking changes are in the `Breaking Changes` section.
- **user-friendly**: add documentation links if necessary.

> - **user-oriented**: do not add changelogs that do not impact the code usage (e.g., documentation, tests, or CI modifications).

Reviewer: ```suggestion
- **user-oriented**: do not add changelogs that do not impact the code (e.g., documentation, tests, or CI modifications).
```

Author: https://github.com/meilisearch/integration-guides/pull/68#discussion_r545055979
